File: Max Consecutive Ones III - Leetcode 1004/brute force.cpp
class Solution {
public:
    int longestOnes(vector<int>& nums, int k) {
        int n = nums.size();
        int maxLen = 0;

        // Iterate over all starting points of subarrays
        for (int i = 0; i < n; ++i) {

            // Check each subarray starting from `i`
            for (int j = i; j < n; ++j) {
                int zeroCount = 0;
                for (int c = i; c <= j; ++c) {
                    if (nums[c] == 0) 
                        zeroCount++;
                }

                // If the number of zeroes exceeds k, 
                // update the maximum length
                if (zeroCount <= k) {
                    maxLen = max(maxLen, j - i + 1);
                }

            }
        }

        return maxLen;
    }
};

File: Max Consecutive Ones III - Leetcode 1004/brute force.py
class Solution(object):
    def longestOnes(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        longest_ones = 0
        
        # Iterate over all starting points
        for i in range(len(nums)):

            # Iterate over all possible subarrays starting from i
            for j in range(i + 1, len(nums) + 1):
                window_len = j - i
                num_zeros = nums[i:j].count(0)  # O(k) operation
                if num_zeros <= k:
                    longest_ones = max(longest_ones, window_len)
        return longest_ones
